File: lab2/python/stopwatch.py
import time


class Stopwatch:
    """
    A simple class for measuring the time it takes to run some code.
    If you want to do some serious timing you should consider using
    the `timeit` module instead.
    """
    _start: float

    def __init__(self):
        self.reset()

    def reset(self):
        """Resets the elapsed time to 0."""
        self._start = time.perf_counter()

    def elapsed_time(self):
        """Returns the elapsed time in seconds after last reset."""
        now = time.perf_counter()
        return now - self._start

    def finished(self, task: str):
        """Prints a timing report and resets the elapsed time."""
        print(f"{task} took {self.elapsed_time():.2f} seconds.")
        self.reset()

File: lab3/java/Stopwatch.java

/**
 * A simple class for measuring the time it takes to run some code.
 * If you want to do some serious timing you should consider using
 * something else instead.
 */
public class Stopwatch {
    private long start;

    public Stopwatch() {
        this.reset();
    }

    /**
     * Resets the elapsed time to 0.
     */
    public void reset() {
        this.start = System.currentTimeMillis();
    }

    /**
     * Returns the elapsed time in seconds after last reset.
     */
    public double elapsedTime() {
        long now = System.currentTimeMillis();
        return (now - this.start) / 1000.0;
    }

    /**
     * Prints a timing report and resets the elapsed time.
     */
   public void finished(String task) {
       System.out.printf("%s took %.2f seconds.\n", task, this.elapsedTime());
       this.reset();
   }

}

